File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.cpp
class Solution
{
public:
    string convert(string s, int numRows)
    {
        if (numRows == 1 || s.length() <= numRows)
            return s;
        vector<string> rows(min(numRows, int(s.size())));
        int curRow = 0, dir = -1;
        for (char c : s)
        {
            rows[curRow] += c;
            if (curRow == 0 || curRow == numRows - 1)
                dir *= -1;
            curRow += dir;
        }
        string ret;
        for (string row : rows)
            ret += row;
        return ret;
    }
};

File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.java
class Solution {
    public String convert(String s, int numRows) {
        if (numRows == 1 || s.length() <= numRows) return s;
        StringBuilder[] rows = new StringBuilder[Math.min(numRows, s.length())];
        for (int i = 0; i < rows.length; i++) rows[i] = new StringBuilder();
        int curRow = 0, dir = -1;
        for (char c : s.toCharArray()) {
            rows[curRow].append(c);
            if (curRow == 0 || curRow == numRows - 1) dir *= -1;
            curRow += dir;
        }
        StringBuilder result = new StringBuilder();
        for (StringBuilder row : rows) result.append(row);
        return result.toString();
    }
}